web/src/ProviderTable.js

```diff
@@ -113,10 +122,38 @@ class ProviderTable extends React.Component {
           return (
             <Switch checked={text} onChange={checked => {
               this.updateField(table, index, 'canSignIn', checked);
+              if (!checked && defaultProvider === record.name) {
+                defaultProvider = null;
+                this.updateField(table, index, 'default', false);
+              }
             }} />
```

Comment:
I think that it's no need to update `defaultProvider`, which is set by application admin and can be empty.

Only need to decide whether to set `default=true` according to ``canSignUp || canSignIn``

Writing in this way will cause the logic of the partial code to be complicated and difficult to maintain.